### PR TITLE
Replace `mir_built` query with a hook and use mir_const everywhere instead

### PR DESCRIPTION
**Pull request by oli-obk**
A small perf improvement due to less dep graph handling.

Mostly just a cleanup to get rid of one of our many mir queries

### CODE REVIEW

**Comment by oli-obk** [comment]
Ah this is just no_hash on mir_const but not mir_built

**Comment by oli-obk** [comment]
r? compiler

File: compiler/rustc_mir_transform/src/lib.rs [hunk 1]
```diff
@@ -282,7 +282,7 @@ fn mir_const_qualif(tcx: TyCtxt<'_>, def: LocalDefId) -> ConstQualifs {
 /// Make MIR ready for const evaluation. This is run on all MIR, not just on consts!
 /// FIXME(oli-obk): it's unclear whether we still need this phase (and its corresponding query).
 /// We used to have this for pre-miri MIR based const eval.
```
Comment (bjorn3) [inline compiler/rustc_mir_transform/src/lib.rs:284]: Is this FIXME still relevant?

File: compiler/rustc_middle/src/query/mod.rs [hunk 2]
```diff
@@ -489,7 +489,7 @@ rustc_queries! {
     /// ready for const qualification.
     ///
     /// See the README for the `mir` module for details.
-    query mir_const(key: LocalDefId) -> &'tcx Steal<mir::Body<'tcx>> {
+    query mir_built(key: LocalDefId) -> &'tcx Steal<mir::Body<'tcx>> {
         desc { |tcx| "preparing `{}` for borrow checking", tcx.def_path_str(key) }
```
Comment (bjorn3) [inline compiler/rustc_middle/src/query/mod.rs:493]: Shouls the description be updated?

**Review by davidtwco — APPROVED** [review]

**Pull request merged**